aplib.net-demo/Assets/Scripts/Player/Weapons/RangedWeapon.cs

```diff
@@ -30,17 +29,19 @@ void Start()
     /// </summary>
     public override void UseWeapon()
     {
-        RaycastHit[] hits = Physics.RaycastAll(CameraManager.Instance.MainCamera.transform.position, CameraManager.Instance.MainCamera.transform.forward, _range);
+        RaycastHit[] hits = Physics.RaycastAll(Camera.main.transform.position, Camera.main.transform.forward, _range);
 
+        // Grabs all objects that collide with this line by order of closest to the player till the furthest the range will let it be.
         IEnumerable<RaycastHit> orderedHits = hits.OrderBy(hit => hit.distance);
 
+        // Will damage only enemies and the ray will stop when it hits an object that is not an enemy.
         foreach (RaycastHit hit in orderedHits)
         {
             if (!hit.collider.CompareTag("Enemy"))
                 break;
 
-            // We need to check if the enemy has an BasicEnemy component before dealing damage
-            BasicEnemy enemy = hit.collider.GetComponent<BasicEnemy>() ?? hit.collider.GetComponentInParent<DummyEnemy>();
+            // Check if the enemy has a DummyEnemy component.
+            DummyEnemy enemy = hit.collider.GetComponentInParent<DummyEnemy>();
             if (enemy != null) enemy.TakeDamage(_damage);
```

Comment:
Don't we want to break after we have hit an enemy? Wouldn't it be possible to hit two enemies back to back? Is this a case of: "It's a feature not a bug" as the bow is piercing.

Reply:
Yes this is a feature.